src/loki/search.cc

```diff
@@ -320,8 +319,10 @@ struct bin_handler_t {
             tangent_angle(index, candidate.point, info.shape(),
                           GetOffsetForHeading(edge->classification(), edge->use()), edge->forward());
         auto layer = info.layer();
-        // do we want this edge
-        if (costing->Allowed(edge, tile, kDisallowShortcut)) {
+        // do we want this edge, note we have to re-evaluate the filter check because we may be
+        // seeing these edges a second time (filtered out before)
+        if (costing->Allowed(edge, tile, kDisallowShortcut) &&
+            !search_filter(edge, *costing, tile, location.search_filter_)) {
```

Comment:
this one and the one right below here are in the `correlate_node` method. this was the primary bug we found in real test cases. node snapping is common in gurka because of.. conventions :smile: